kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/admin/impl/UserTaskAdminServicesClientImpl.java

```diff
@@ -55,179 +55,286 @@ public UserTaskAdminServicesClientImpl(KieServicesConfiguration config, ClassLoa
 
     @Override
     public void addPotentialOwners(String containerId, Long taskId, boolean removeExisting, OrgEntities orgEntities) {
+        addPotentialOwners(null, containerId, taskId, removeExisting, orgEntities);
+    }
+
+    @Override
+    public void addPotentialOwners(String userId, String containerId, Long taskId, boolean removeExisting, OrgEntities orgEntities) {
         if( config.isRest() ) {
             Map<String, Object> valuesMap = new HashMap<String, Object>();
             valuesMap.put(CONTAINER_ID, containerId);
             valuesMap.put(TASK_INSTANCE_ID, taskId);
 
             Map<String, String> headers = new HashMap<String, String>();
             String queryString = "?remove=" + removeExisting;
+            if(BYPASS_AUTH_USER && userId != null) {
+                queryString += "&user=" + userId;
+            }
 
             makeHttpPutRequestAndCreateCustomResponse(
                     build(loadBalancer.getUrl(), ADMIN_TASK_URI + "/" + TASK_INSTANCE_POT_OWNERS_USERS_URI, valuesMap) + queryString, orgEntities, null, headers);
         } else {
+            Object[] argvs = new Object[]{containerId, taskId, removeExisting};
+            if(BYPASS_AUTH_USER && userId != null) {
+                argvs = new Object[]{userId, containerId, taskId, removeExisting};
```

Comment:
JMS messages sent by client are converted to appropriate base classes, for example for `UserTaskAdminService` there is triggered `UserTaskAdminServiceBase` instance. This instance has only a method containing `userId` parameter. So some value needs to be provided as `userId` for the JMS in all cases.

Same for other cases below.

Reply:
the other way round... old calls need to come back for backward compatibility.